src/cloud-cli/userdb.ts

```diff
@@ -0,0 +1,105 @@
+import axios from "axios";
+import { createGlobalLogger } from "../telemetry/logs";
+import { getCloudCredentials } from "./utils";
+
+export async function createUserDb(host: string, port: string, dbName: string, adminName: string, adminPassword: string, sync: boolean) {
+  const logger = createGlobalLogger();
+  const userCredentials = getCloudCredentials();
+  const bearerToken = "Bearer " + userCredentials.token;
+  
+  try {
+    await axios.post(`http://${host}:${port}/${userCredentials.userName}/databases/userdb`, 
+    {"Name": dbName,"AdminName": adminName, "AdminPassword": adminPassword},
+    {
+      headers: {
+        "Content-Type": "application/json",
+        Authorization: bearerToken,
+      },
+    });
+
+    logger.info(`Successfully started creating database: ${dbName}`);
+    let status = ""
+    if(sync) {
+
+      while (status != "available") {
+        await sleep(60000)
+        // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
+        const data = await getDb(host, port, dbName)
+        // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
+        logger.info(data)
+        // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment, @typescript-eslint/no-unsafe-member-access
+        status = data.Status
+      }
+
+    }
+  } catch (e) {
+    if (axios.isAxiosError(e) && e.response) {
+      logger.error(`Error creating database ${dbName}: ${e.response?.data}`);
+    } else {
+      logger.error(`Error creating database ${dbName}: ${(e as Error).message}`);
+    }
+  }
+}
+
+export async function deleteUserDb(host: string, port: string, dbName: string) {
```

Comment:
Is it possible to make delete optionally synchronous as well? For example, I may want to wait until the DB is actually deleted from RDS.

Reply:
fixed